<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Pre-processing â€¢ CellFunMap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Pre-processing">
<meta property="og:description" content="CellFunMap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellFunMap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CellFunMap.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pre-processing.html">Pre-processing</a>
    </li>
    <li>
      <a href="../articles/visualization_gsea.html">Visualization of GSEA Result</a>
    </li>
    <li>
      <a href="../articles/topicmodel_visualization.html">Topic modeling and visualization</a>
    </li>
    <li>
      <a href="../articles/transfer_topicmodel.html">Transfer topic model and visualization</a>
    </li>
    <li>
      <a href="../articles/transfer_annotation.html">Transfer annotation from reference to query data</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Visualize in Built-in Shiny APP</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xiaobaicainiao666/CellFunMap/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pre-processing_files/header-attrs-2.14/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Pre-processing</h1>
            
            <h4 class="date">2022-08-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/xiaobaicainiao666/CellFunMap/blob/master/vignettes/pre-processing.Rmd"><code>vignettes/pre-processing.Rmd</code></a></small>
      <div class="hidden name"><code>pre-processing.Rmd</code></div>

    </div>

    
    
<p>The <strong>CellFunMap</strong> package provides a convenient
workflow for data pre-processing (quality control, normalization,
dimension reduction, clustering, differential expression analysis, etc.)
by integrating methods of <code>Seurat</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/xiaobaicainiao666/CellFunMap">CellFunMap</a></span><span class="op">)</span></code></pre></div>
<div id="load-in-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-in-the-data" class="anchor"></a>Load in the data</h2>
<p>CellFunMap allows various types of input, including CellRanger
outputs, raw gene expression matrix, and popular R/python objects used
for single cell analysis such as <code>SingleCellExperiment</code>,
<code>Seurat</code>, <code>CellDataSet</code>, <code>AnnData</code>.
CellFunMap will transform different input data into a
<code>Seurat</code> object.</p>
<p>Let us use <code>pbmc3k.final</code> as a toy example.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'pbmc3k.final'</span>, package <span class="op">=</span> <span class="st">"pbmc3k.SeuratData"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">pbmc3k.final</span><span class="op">)</span>

<span class="va">SeuratObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readData.html">readData</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">pbmc3k.final</span>, type <span class="op">=</span> <span class="st">'Seurat'</span>, species <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">pbmc3k.final</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/gc.html">gc</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="standard-pre-processing-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#standard-pre-processing-workflow" class="anchor"></a>Standard pre-processing workflow</h2>
<p>Then we can perform pre-processing conveniently. If users provide a
pre-processed data, this step can be skipped.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SeuratObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CalMTpercent.html">CalMTpercent</a></span><span class="op">(</span><span class="va">SeuratObj</span>, by <span class="op">=</span> <span class="st">"use_internal_data"</span><span class="op">)</span>
<span class="va">SeuratObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCfun.html">QCfun</a></span><span class="op">(</span><span class="va">SeuratObj</span>, plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="co"># set plot = T if you want to get the QC reports.</span>
<span class="va">SeuratObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunSeurat.html">RunSeurat</a></span><span class="op">(</span><span class="va">SeuratObj</span>, nPCs <span class="op">=</span> <span class="fl">10</span>, resolution <span class="op">=</span> <span class="fl">1</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Idents.html">Idents</a></span><span class="op">(</span><span class="va">SeuratObj</span><span class="op">)</span><span class="op">)</span> <span class="co"># see how many clusters we got.</span></code></pre></div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-set-enrichment-analysis-gsea" class="anchor"></a>Gene Set Enrichment Analysis (GSEA)</h2>
<p>Then we perform Gene Set Enrichment Analysis (GSEA) on the
<code>Seurat</code> object. CellFunMap provides a encapsulated function
for GSEA with different databases such as GO (Gene Ontology), KEGG,
Reactome, MSigDb, WikiPathways, DO, NCG, DGN.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SeuratObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunGSEA.html">RunGSEA</a></span><span class="op">(</span><span class="va">SeuratObj</span>, by <span class="op">=</span> <span class="st">'GO'</span><span class="op">)</span>

<span class="co"># save object for later use.</span>
<span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span><span class="op">(</span><span class="va">SeuratObj</span>, file <span class="op">=</span> <span class="st">"SeuratObj.RData"</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Shanni Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
