<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ CellFunTopic</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellFunTopic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/CellFunTopic.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/pre-processing.html">Pre-processing</a>
    </li>
    <li>
      <a href="../articles/visualization_gsea.html">Visualization of GSEA Result</a>
    </li>
    <li>
      <a href="../articles/topicmodel_visualization.html">Topic modeling and visualization</a>
    </li>
    <li>
      <a href="../articles/transfer_topicmodel.html">Transfer topic model and visualization</a>
    </li>
    <li>
      <a href="../articles/transfer_annotation.html">Transfer annotation from reference to query data</a>
    </li>
    <li>
      <a href="../articles/shiny_app.html">Visualize in Built-in Shiny APP</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xiaobaicainiao666/CellFunTopic/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-pre-processing" class="hasAnchor"><a href="#section-pre-processing" class="anchor"></a>Pre-processing</h2>
          <p class="section-desc"><p>Functions used for pre-processing.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="readData.html">readData()</a></code> </p>
        </td>
        <td><p>Load data from different sources</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CalMTpercent.html">CalMTpercent()</a></code> </p>
        </td>
        <td><p>Calculate the proportion of transcripts mapping to mitochondrial genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="DetectGeneIDtype.html">DetectGeneIDtype()</a></code> </p>
        </td>
        <td><p>Detect the gene ID type of Seurat object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="QCfun.html">QCfun()</a></code> </p>
        </td>
        <td><p>quality control according to 'nCount_RNA', 'nFeature_RNA', 'percent.mt', 'CellsPerGene'</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RunSeurat.html">RunSeurat()</a></code> </p>
        </td>
        <td><p>basic single cell analysis with Seurat</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RunGSEA.html">RunGSEA()</a></code> </p>
        </td>
        <td><p>Perform Gene Set Enrichment Analysis (GSEA) on Seurat object</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-visualization-of-gsea" class="hasAnchor"><a href="#section-visualization-of-gsea" class="anchor"></a>Visualization of GSEA</h2>
          <p class="section-desc"><p>Functions used for visualization of GSEA Result.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="circleplot.html">circleplot()</a></code> </p>
        </td>
        <td><p>circle plot of clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clustercorplot.html">clustercorplot()</a></code> <code><a href="clustercorplot.html">clustercorplot_jaccard()</a></code> </p>
        </td>
        <td><p>circle plot of clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Cosine_networkByGSEA.html">Cosine_networkByGSEA()</a></code> </p>
        </td>
        <td><p>Network showing cosine similarity between clusters according to GSEA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="edge_bundling_GSEA.html">edge_bundling_GSEA()</a></code> </p>
        </td>
        <td><p>Hierarchical edge bundling plots helps visualizing correlation or similarity between clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="embeddedplot.html">embeddedplot()</a></code> </p>
        </td>
        <td><p>show embedded histogram or pie chart on UMAP/TSNE plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GOboxplot.html">GOboxplot()</a></code> </p>
        </td>
        <td><p>boxplot showing enrichment score of child or parent GOs of specific GO</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GOcircleplot.html">GOcircleplot()</a></code> </p>
        </td>
        <td><p>show pathways and genes in chord diagram</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gseaHeatmap.html">gseaHeatmap()</a></code> </p>
        </td>
        <td><p>Heatmap of GSEA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hierarchyplot_tree.html">hierarchyplot_tree()</a></code> </p>
        </td>
        <td><p>show relationship between clusters and pathways</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pathwayScatterplot.html">pathwayScatterplot()</a></code> </p>
        </td>
        <td><p>Scatter plot showing pathway enrichment score</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ID2Description.html">ID2Description()</a></code> </p>
        </td>
        <td><p>Get pathway IDs and their corresponding description</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simplifyEnrichmentplot.html">simplifyEnrichmentplot()</a></code> </p>
        </td>
        <td><p>cluster functional terms into groups by clustering the similarity matrix of the terms</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scPalette.html">scPalette()</a></code> </p>
        </td>
        <td><p>Generate colors from a customed color palette</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scPalette2.html">scPalette2()</a></code> </p>
        </td>
        <td><p>Generate colors from a customed color palette</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-topic-modeling-and-visualization" class="hasAnchor"><a href="#section-topic-modeling-and-visualization" class="anchor"></a>Topic modeling and visualization</h2>
          <p class="section-desc"><p>Functions used for topic modeling and visualization.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="runLDA.html">runLDA()</a></code> </p>
        </td>
        <td><p>Run topic modeling with GSEA result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cluster_topic_hmp.html">cluster_topic_hmp()</a></code> </p>
        </td>
        <td><p>Heatmap showing probability between topics and clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_sankey.html">plot_sankey()</a></code> </p>
        </td>
        <td><p>Sankey diagram showing the best assigned topic of each cluster</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosineheatmap.html">cosineheatmap()</a></code> </p>
        </td>
        <td><p>Heatmap showing cosine similarity between topics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosinehmp_cluster.html">cosinehmp_cluster()</a></code> </p>
        </td>
        <td><p>Heatmap showing cosine similarity between clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine_network_cluster.html">cosine_network_cluster()</a></code> </p>
        </td>
        <td><p>Network showing cosine similarity between clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine_network_term.html">cosine_network_term()</a></code> </p>
        </td>
        <td><p>Network of cosine similarity between terms</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="topicProb.html">topicProb()</a></code> </p>
        </td>
        <td><p>Show topic activity on UMAP/TSNE cell map</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Topterms_Topic.html">Topterms_Topic()</a></code> </p>
        </td>
        <td><p>Show top terms of each topic</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hist_cluster_topic.html">hist_cluster_topic()</a></code> </p>
        </td>
        <td><p>barplot to show probability of assigned topics in clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hist_topic_term.html">hist_topic_term()</a></code> </p>
        </td>
        <td><p>barplot to show top terms of each topic</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="topicNW1.html">topicNW1()</a></code> </p>
        </td>
        <td><p>network showing top terms of topics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="topicNW2.html">topicNW2()</a></code> </p>
        </td>
        <td><p>network showing top terms of topics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="topicNW3.html">topicNW3()</a></code> </p>
        </td>
        <td><p>echart4r network showing top terms of topics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="umap_cluster.html">umap_cluster()</a></code> </p>
        </td>
        <td><p>UMAP on cluster-topic probability matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="wordcloud_topic.html">wordcloud_topic()</a></code> <code><a href="wordcloud_topic.html">wordcloud_topic_3D()</a></code> </p>
        </td>
        <td><p>word cloud of top terms of topic</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lda.rep.html">lda.rep()</a></code> </p>
        </td>
        <td><p>calculate metrics to find the best number of topics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotindex.html">plotindex()</a></code> </p>
        </td>
        <td><p>plot metrics to find the best number of topics</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-transfer-topic-model" class="hasAnchor"><a href="#section-transfer-topic-model" class="anchor"></a>Transfer topic model</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="transfer_LDA.html">transfer_LDA()</a></code> </p>
        </td>
        <td><p>Transfer LDA model from reference data to query data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine_heatmap2.html">cosine_heatmap2()</a></code> </p>
        </td>
        <td><p>Heatmap showing cosine similarity or pearson correlation between reference data and query data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine_hive.html">cosine_hive()</a></code> </p>
        </td>
        <td><p>Hive diagram showing cosine similarity between reference data and query data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cosine_network_cluster2.html">cosine_network_cluster2()</a></code> </p>
        </td>
        <td><p>network showing cosine similarity between reference data and query data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sankey_comparison.html">sankey_comparison()</a></code> </p>
        </td>
        <td><p>Sankey diagram comparing the reference LDA model and query data</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-transfer-annotation" class="hasAnchor"><a href="#section-transfer-annotation" class="anchor"></a>Transfer annotation</h2>
          <p class="section-desc"><p>Functions used for annotation transfer from reference to query
data.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="oversample_ref.html">oversample_ref()</a></code> </p>
        </td>
        <td><p>Oversample the reference data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="predictFun.html">predictFun()</a></code> </p>
        </td>
        <td><p>Predict cell types of query data based on reference data</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-built-in-shiny-app" class="hasAnchor"><a href="#section-built-in-shiny-app" class="anchor"></a>Built-in Shiny APP</h2>
          <p class="section-desc"><p>Visualize analysis result in Built-in Shiny APP.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="visualize_in_shiny.html">visualize_in_shiny()</a></code> </p>
        </td>
        <td><p>visualize the Seurat object in a built-in shiny app</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data" class="hasAnchor"><a href="#section-data" class="anchor"></a>Data</h2>
          <p class="section-desc"><p>Internal data.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="GO2level.html">GO2level</a></code> </p>
        </td>
        <td><p>GO terms and their corresponding level or ontology.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="HCL_ldaOut.html">HCL_ldaOut</a></code> </p>
        </td>
        <td><p>built-in reference topic model derived from HCL.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MCA_ldaOut.html">MCA_ldaOut</a></code> </p>
        </td>
        <td><p>built-in reference topic model derived from MCA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MTGENE.html">MTGENE</a></code> </p>
        </td>
        <td><p>Mitochondrial genes of 3 species.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="txid_gene_MT.html">txid_gene_MT</a></code> </p>
        </td>
        <td><p>Mitochondrial genes of all species.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="txid_species_gene_MT.html">txid_species_gene_MT</a></code> </p>
        </td>
        <td><p>Mitochondrial genes of all species.</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Shanni Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


